modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/CommonComponentBeanDefinitionCreator.java

```diff
@@ -0,0 +1,139 @@
+    processObjectConstructionParameters(springComponentModels, component, component, request,
+                                        new BeanDefinitionBuilderHelper(beanDefinitionBuilder));
+    processMuleProperties(component, beanDefinitionBuilder, beanDefinitionPostProcessor);
+    if (componentBuildingDefinition.isPrototype()) {
+      beanDefinitionBuilder.setScope(SPRING_PROTOTYPE_OBJECT);
+    }
+    AbstractBeanDefinition originalBeanDefinition = beanDefinitionBuilder.getBeanDefinition();
+    AbstractBeanDefinition wrappedBeanDefinition = adaptBeanDefinition(originalBeanDefinition);
+    if (originalBeanDefinition != wrappedBeanDefinition) {
+      request.getSpringComponentModel().setType(wrappedBeanDefinition.getBeanClass());
+    }
+    request.getSpringComponentModel().setBeanDefinition(wrappedBeanDefinition);
+  }
+
+  static void processMuleProperties(ComponentAst component, BeanDefinitionBuilder beanDefinitionBuilder,
+                                    BeanDefinitionPostProcessor beanDefinitionPostProcessor) {
+    if (component == null) {
+      return;
+    }
+
+    // for now we skip custom-transformer since requires injection by the object factory.
+    if (beanDefinitionPostProcessor != null && beanDefinitionPostProcessor.getGenericPropertiesCustomProcessingIdentifiers()
+        .contains(component.getIdentifier())) {
+      return;
+    }
+    component.directChildrenStream()
+        .filter(innerComponent -> {
+          ComponentIdentifier identifier = innerComponent.getIdentifier();
+          return identifier.equals(MULE_PROPERTY_IDENTIFIER)
+              || identifier.equals(MULE_PROPERTIES_IDENTIFIER);
+        })
+        .forEach(propertyComponentModel -> {
+          Pair<String, Object> propertyValue = getPropertyValueFromPropertyComponent(propertyComponentModel);
+          beanDefinitionBuilder.addPropertyValue(propertyValue.getFirst(), propertyValue.getSecond());
+        });
+  }
+
+  private AbstractBeanDefinition adaptBeanDefinition(AbstractBeanDefinition originalBeanDefinition) {
+    Class beanClass;
+    if (originalBeanDefinition instanceof RootBeanDefinition) {
+      beanClass = ((RootBeanDefinition) originalBeanDefinition).getBeanClass();
+    } else {
+      try {
+        beanClass = originalBeanDefinition.getBeanClass();
+      } catch (IllegalStateException e) {
+        try {
+          beanClass = org.apache.commons.lang3.ClassUtils.getClass(originalBeanDefinition.getBeanClassName());
+        } catch (ClassNotFoundException e2) {
+          throw new RuntimeException(e2);
+        }
+      }
+    }
+
+    BeanDefinition newBeanDefinition;
+    if (areMatchingTypes(SecurityFilter.class, beanClass)) {
+      newBeanDefinition = rootBeanDefinition(SecurityFilterMessageProcessor.class)
+          .addConstructorArgValue(originalBeanDefinition)
+          .getBeanDefinition();
+      return (AbstractBeanDefinition) newBeanDefinition;
+    } else {
+      return originalBeanDefinition;
+    }
```

Comment:
Inverting this if might make the code clearer

Reply:
i think it is good now with the other changes